### PR TITLE
Movie query

### PR DESCRIPTION
**Pull request by alexscheitlin**
### Description

This PR adds a query to fetch a specific or random movie with data from both the metadata and poster service. See the issue for more detailed information.

### Related Issues

#27 

### Checklist

- [x] Have you added tests where necessary? Do all the test pass? 
- [x] Have you added descriptive comments to your code?
- [x] Have you updated the documentation related to this propo

### CODE REVIEW

**Comment by rschlaefli** [comment]
It will be an additional challenge to do the E2E test here, as we need both microservices running. Have a look at https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices

It should not be too hard to use the built images and run the latest dev images as a service

**Comment by alexscheitlin** [comment]
> It will be an additional challenge to do the E2E test here, as we need both microservices running. Have a look at https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices
> 
> It should not be too hard to use the built images and run the latest dev images as a service

I've added both services using the latest images from docker hub. However the e2e test still fails with the same error. What am I missing?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -6,6 +6,16 @@ jobs:
   test:
     runs-on: ubuntu-latest
 
+    services:
+      metadata:
+        image: kwizapp/metadata-service:latest
+        ports:
+          - 8004:8004
```
Comment (rschlaefli) [inline .github/workflows/main.yml]:
```suggestion
          - 8004:3000
```

File: src/movie/movie.service.ts [hunk 2]
```diff
@@ -1,25 +1,58 @@
...
+  POSTER_SERVICE = process.env.POSTER_SERVICE_URL
 
-  getMovieByIndex(index: number) {
+  async getMovieByImdbId(imdbId: string): Promise<Movie> {
     try {
-      return movies[index]
-    } catch (e) {
-      return null
+      // get metadata
+      const metadataUrl = `${this.METADATA_SERVICE}?id=${imdbId}`
```
[the remark this answers is not in the document]
Comment (alexscheitlin) [reply src/movie/movie.service.ts]: This is already changed and commited in the upcoming PRs that depend on this one.

File: src/movie/movie.utils.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { Movie } from 'src/graphql'
+
+export const extractDataFromMetadataResponse = (response): Movie => {
+  const movie = response.data[0]
+
+  return {
+    imdbId: movie.imdb_id,
+    title: movie.title,
+    releaseYear: parseInt(movie.release_date.substr(0, 4)),
```
[the remark this answers is not in the document]
Comment (alexscheitlin) [reply src/movie/movie.utils.ts]: This is already changed and commited in the upcoming PRs that depend on this one.

File: .github/workflows/main.yml (hunk 1 above)
Comment (alexscheitlin) [reply .github/workflows/main.yml]: As discussed, we won't do this.

File: src/movie/movie.utils.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    posterPath: '',
+  }
+}
+
+export const extractDataFromPosterResponse = (response): Movie => {
+  return {
+    imdbId: '',
```
[the remark this answers is not in the document]
Comment (alexscheitlin) [reply src/movie/movie.utils.ts]: Nice, I did not know that something like a `Partial` exists. Thanks for the suggestion!
I've created a separate issue to introduce this change (#65).

**Pull request merged**